How-Authentication-Works-in-TeamsFx.md

```diff
@@ -0,0 +1,174 @@
+You can find the template of auth-start.html [here](https://github.com/OfficeDev/TeamsFx/blob/main/templates/tab/js/default/public/auth-start.html) for JavaScript and [here](https://github.com/OfficeDev/TeamsFx/blob/main/templates/tab/ts/default/public/auth-start.html) for TypeScript.
+
+4. After login in the AAD login page, user will be redirected to auth-end.html. In auth-end.html, will use [msal](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#redirect-apis) to use Auth Code Flow with PKCE. Since msal use the session storage for [cache](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/caching.md), will return session storage to SDK. The scheme of response will be:
+      ```
+      {
+        "sessionStorage": {
+          "key": "value",
+          ...
+        },
+        ...
+      }
+      ``` 
+You can find the template of auth-end.html [here](https://github.com/OfficeDev/TeamsFx/blob/main/templates/tab/js/default/public/auth-end.html) for JavaScript and [here](https://github.com/OfficeDev/TeamsFx/blob/main/templates/tab/ts/default/public/auth-end.html) for TypeScript.
+
+5. TeamsFx SDK will get the response from the auth-end page, and will:
+      * If the response is empty or can not parse to JSON object, will return error.
+      * If "code" exists in the response, will return error. You can check [here](#how-to-solve-the-error-found-auth-code-in-response-auth-code-is-not-support-for-current-version-of-sdk) for detail.
+      * If "sessionStorage" exists in the response, will set the key value pairs to the session storage.
+
+
+6. After this, the login flow has been completed. You can use the following code to call Graph API: 
+    ```ts
+    const graphClient = createMicrosoftGraphClientWithCredential(teamsUserCredential, ["User.Read"]); 
+    const profile = await graphClient.api("/me").get();
+    ```
+
+### Get Token Flow
+When getting token flow is triggered, Teams SDK will first try to use msal to check whether Access Token exists in session storage. Here msal will handle the scenario that the token exists and is expired. If succeeded, return Access Token. If failed to get Access Token from session storage, Teams SDK will try to use msal to launch silent SSO. If succeeded, return Access Token. If failed in silent SSO, you need to trigger the login flow to get Access Token.
+
+
+
+## How On-Behalf-Of flow works in Teams Tab with Azure Function backend
+If your Teams Tab App has an backend service such as Azure Function application, and you want to access current user information protected by Azure Active Directory in backend service, for example, if your Azure Function needs to call Microsoft Graph APIs to obtain a user's profile photo, then you need to authenticate the user with On-Behalf-Of flow to retrieve the appropriate authentication tokens.
```

Comment:
Add link to obo like auth code flow in line 9

Reply:
Sure, thanks~

Reply:
Updated